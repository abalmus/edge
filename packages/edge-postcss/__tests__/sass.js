import { compile } from "./core"

// ====================================================
// ================ SASS INSPIRED =====================
// ====================================================

test("Sassy Mixins", () =>
  compile("@mixin simple{ color: red; } h1 { @include simple; }")
)

test("Sassy Variables", () =>
  compile("$bgColor: red; h1 { background: $bgColor; }")
)

test("Nested Basic", () =>
  compile("body { h1 { font-weight: bold; } h2 { font-weight: normal; }}")
)

test("Nested Parent Selector", () =>
  compile("body { ul { li { &:first-child { margin-top: 0; }}}}")
)

test("Calc Simple", () =>
  compile("h1 { width: calc(10% + 3%); }")
)

test("Calc Complex", () =>
  compile("h1 { width: calc(99.999% * 3 / 10 - 22px); }")
)
