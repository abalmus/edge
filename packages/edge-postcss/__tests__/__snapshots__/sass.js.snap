// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calc Complex 1`] = `
"h1{
width:calc(29.9997% - 22px)
}"
`;

exports[`Calc Simple 1`] = `
"h1{
width:13%
}"
`;

exports[`Nested Basic 1`] = `
"body h1{
font-weight:700
}

body h2{
font-weight:400
}"
`;

exports[`Nested Parent Selector 1`] = `
"body ul li:first-child{
margin-top:0
}"
`;

exports[`Sassy Mixins 1`] = `
"h1{
color:#ff4136
}"
`;

exports[`Sassy Variables 1`] = `
"h1{
background:#ff4136
}"
`;
